xlators/cluster/dht/src/dht-rename.c

```diff
@@ -123,7 +123,9 @@ dht_rename_dir_cbk(call_frame_t *frame, void *cookie, xlator_t *this,
     local = frame->local;
     prev = cookie;
     subvol_cnt = dht_subvol_cnt(this, prev);
-    local->ret_cache[subvol_cnt] = op_ret;
+
+    if (subvol_cnt >= 0)
+        local->ret_cache[subvol_cnt] = op_ret;
```

Comment:
Isn't this already present in the devel branch ?

Reply:
This is from another patch it shouldn't be here.

Reply:
Yes, I'm based on another tree with this fix. Let me rebase to devel if that's in.